service-deployment/src/grafana/start_server.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+chown -R grafana:grafana /var/lib/grafana /var/log/grafana
+
+if [ ! -z ${GF_AWS_PROFILES+x} ]; then
+    for profile in ${GF_AWS_PROFILES}; do
+        access_key_varname="GF_AWS_${profile}_ACCESS_KEY_ID"
+        secret_key_varname="GF_AWS_${profile}_SECRET_ACCESS_KEY"
+        region_varname="GF_AWS_${profile}_REGION"
+
+        if [ ! -z "${!access_key_varname}" -a ! -z "${!secret_key_varname}" ]; then
+            echo "[${profile}]" >> ~grafana/.aws/credentials
+            echo "aws_access_key_id = ${!access_key_varname}" >> ~grafana/.aws/credentials
+            echo "aws_secret_access_key = ${!secret_key_varname}" >> ~grafana/.aws/credentials
+            if [ ! -z "${!region_varname}" ]; then
+                echo "region = ${!region_varname}" >> ~grafana/.aws/credentials
+            fi
+        fi
+    done
+
+    chown grafana:grafana -R ~grafana/.aws
+    chmod 600 ~grafana/.aws/credentials
+fi
+
+# upgrade all installed plugins
+if [ "$UPGRADEALL" = true ] ; then
+    grafana-cli --pluginsDir "${GF_PLUGIN_DIR}" plugins upgrade-all || true
```

Comment:
>GF_PLUGIN_DIR [](start = 32, length = 13)

where to get this value?

Reply:
It is defined in dockerfile

Reply:
can you list all env. variables defined in dockerfile in the beginning? so people can easily figure out where it comes from. @ydye, do we have better practice ?

---
In reply to: [157507719](https://github.com/Microsoft/pai/pull/102#discussion_r157507719) [](ancestors = 157507719)

Reply:
Define Env in kubernets' YAML file will be better. And decouple the env from docker image will be better. @fanyangCS